repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/services/httpconnectionpool/HttpsClientAuthNonAuthedOriginTest.groovy

```diff
@@ -0,0 +1,56 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.services.httpconnectionpool
+
+import framework.ReposeValveTest
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+
+/**
+ * Created by adrian on 3/3/17.
+ */
+class HttpsClientAuthNonAuthedOriginTest extends ReposeValveTest {
+
+    def setup() {
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort)
+    }
+
+    def cleanup() {
+        if (deproxy)
+            deproxy.shutdown()
+    }
+
+    def "repose should have no problems with a non authed origin, while connection pool has auth configured"() {
+        given:
+        def params = properties.getDefaultTemplateParams()
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/services/httpconnectionpool/clientauth/nonauthedorigin", params)
+        repose.start()
+        waitUntilReadyToServiceRequests()
```

Comment:
The other thing about setting up Repose here is that it isn't necessarily cleaned up appropriately. Again, this this case, there is just one test and its using the `repose` instance created in `ReposeValveTest`, so it works fine. Still...